### DISCUSSION TITLE
SSG behavior

### DISCUSSION THREAD

**Discussion by railty** [vikejs/vike#1687] [Discussion]
I tried SSG on a brand new scaffold vike app, the generated files are index.html and index.pageContext.json. 

Is this by design? how is this affect SEO

I expect a html with the content embeded, not dynamic loaded. 

Steps:
1. npm create vike@latest to create a react js app.
2. change vire.config.js to plugins: [react(), vike({prerender: true})]
3. create a js file "pages/star-wars/index/+onBeforePrerenderStart.js", which is 
`
export { onBeforePrerenderStart }
 
function minimize(movies) {
    return movies.map((movie) => {
      const { title, release_date, id } = movie
      return { title, release_date, id }
    })
  }
  
async function onBeforePrerenderStart() {
    const response = await fetch('https://brillout.github.io/star-wars/api/films.json')
    const moviesData = await response.json()
    const movies = minimize(moviesData)
    //console.log(movies);
    const moviePageURLs = movies.map(movie => '/star-wars/' + movie.id)
    console.log(moviePageURLs);
  return moviePageURLs
}
 `

4. npm run build

and I got `
 ls dist\client\star-wars\1

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----         6/10/2024  11:20 PM           2169 index.html
-a----         6/10/2024  11:20 PM            410 index.pageContext.json
`

**Comment by railty** [comment]
Anyone? Let me clarify my question. 

1. With my setup above, I am getting HTML files with JavaScript that dynamically load a JSON file. Am I doing this correctly?
2. this is not SSO-friendly, compared to static HTML files with embedded content, correct?

Thanks

**Reply by brillout** [reply]
If the content isn't in the HTML then you're doing some quite unusual.

If you believe it's a Vike bug then feel free to provide a reproduction that is minimal.

**Comment by railty** [comment]
I agree, it seems I did something wrong, can amyone give a tutoral on setting up SSG? 

I basicly did

1. npm create vike@latest to create a react js app.
2. change vire.config.js to plugins: [react(), vike({prerender: true})]
3. npm run build

> `vike v0.4.176 pre-rendering HTML...
> dist/client/index.pageContext.json
> dist/client/about/index.html
> dist/client/about/index.pageContext.json

`

I don't think "dist/client/about/index.pageContext.json" should be there, it should all in "dist/client/about/index.html"

What I missed?

**Reply by brillout** [reply]
Have you seen https://vike.dev/pre-rendering?
